Core/Core/Studio/Model/StudioAPIAuthInteractor.swift

```diff
@@ -20,12 +20,16 @@ import Combine
 import CombineExt
 import WebKit
 
-public enum StudioAPIAuthError: Error {
+public enum StudioAPIAuthError: String, Error {
     case failedToGetLTIs
     case failedToGetTokenFromWebView
     case failedToGetAPIBaseURL
     case studioLTINotFound
     case failedToGetLaunchURL
+
+    var localizedDescription: String {
+        "\(Self.self).\(self)"
+    }
```

Comment:
I would rather call this `stringValue` or something because localizedDescription is usually a full sentence that could be even used for user facing error handling.

Reply:
Good thinking. The main reason I chose this is because every Error has the `localizedDescription` field so when you catch unknown error types you are still safe to use this field. If we create a custom property you have to know if this error type will be thrown or not at a higher level and check for its type so you can call the error's custom property. In this case this error is just one of the many types that can be caught in `CourseSyncStudioMediaInteractorLive`. I think we are on the safe side regarding not putting this error in front of users since we rarely put errors directly in front of them mainly because `localizedDescription`s are awfully formatted and not user friendly.
What I could come up with is an error protocol that has an `analyticsDescription` or something similar so handling of all the different types would be reduced to one but we would still be required to add this custom logic to every place where we want to extract the error's description.